agent/websockets/shim.go

```diff
@@ -410,21 +419,11 @@ func createShimChannel(ctx context.Context, host, shimPath string, rewriteHost b
 }
 
 // Proxy creates a reverse proxy that inserts websocket-shim code into all HTML responses.
-func Proxy(ctx context.Context, wrapped *httputil.ReverseProxy, host, shimPath string, injectShimCode, rewriteHost bool) (http.Handler, error) {
-	var templateBuf bytes.Buffer
-	if err := shimTmpl.Execute(&templateBuf, &struct{ ShimPath string }{ShimPath: shimPath}); err != nil {
-		return nil, err
-	}
-	shimCode := templateBuf.String()
-	if injectShimCode {
-		wrapped.ModifyResponse = func(resp *http.Response) error {
-			return shimBody(resp, shimCode)
-		}
-	}
+func Proxy(ctx context.Context, wrapped http.Handler, host, shimPath string, rewriteHost bool, openWebsocketWrapper func(http.Handler) http.Handler) (http.Handler, error) {
```

Comment:
Please add godoc for the `openWebsocketWrapper` parameter explaining what it is used for and what it needs to do.

Reply:
Done